### DISCUSSION TITLE
Devices connected to wifi range extender not being discovered..

### DISCUSSION THREAD

**Discussion by redstormsju** [netalertx/NetAlertX#576] [Q&A]
Hello all. I am running pi.alert in docker on my Synology nas. After installing it pi.alert discovered all of the devices that are directly connected to my router - wired and wifi!  One of those devices is a TP Link Range extender with its own static ip address. I can see it in the 'Devices' list in pi.alert. 

What I don't see are the devices that are directly connecting to the extender - 5 of them. However, if I click at TP Link within the devices section, then I click on sessions, I can see the devices connected to the extender there. I do not see them under the 'Devices' page of pi.alert. 

Anyone know how I should set up pi.alert to also scan and list those devices under 'Devices'?  Thank you!

**Comment by jokob-sk** [comment]
Hi there,

I don't have an extender myself so I can't try to reproduce this on my end.

You could also consider supplementing the arp-scan with other scanners, such as Pi-Hole (if you use it), Unifi (if available), SNMP, or DHCP.leases files. A lot of routers support SNMP, so that's something I would recommend to look into. 

Hope this helps,
j

**Comment by redstormsju** [comment]
Thank you. I will try these suggestions.

Could I trouble you with another question…

I have a device that shows offline, however I know it’s online as I can
confirm through the device itself.  Under settings for this device in
pi.alert, I ran a manual nmap scan and I get the following message that I
don’t known how to follow:

“Note/ Host seems down. If it is really up, but blocking our ping process,
try -Pn
Nmap done: 1 IP address (0 hosts up) scanned in 3.05 seconds”

Any suggestions or clarification on what I am seeing will be appreciated!

**Reply by jokob-sk** [reply]
Hey @redstormsju ,

The arp-scan relies on the arp-cache which is not always up to date. Combining arp-scan with other methods can give more accurate results. More robust discovery is planned for some future work, e.g. #562 or #434. Let me know if you have suggestions how to verify a device has a specific state. 

Thanks in advance,
j
